### PR TITLE
Add docs for mark-move GC

### PR DESCRIPTION
**Pull request by binier**

### CODE REVIEW

File: storage/src/backend/kv_store_gced.rs [hunk 1]
```diff
@@ -147,14 +163,22 @@ impl<T: 'static + KVStore + Default> KVStore for KVStoreGCed<T> {
...
+    /// Marks an entry as "reused" in the current cycle.
+    /// 
+    /// Entries that are reused/referenced in current cycle
+    /// will be preserved after garbage collection.
     fn mark_reused(&mut self, key: EntryHash) {
         let _ = self.msg.lock().unwrap().send(CmdMsg::MarkReused(key));
     }
 
-    fn retain(&mut self, pred: HashSet<EntryHash>) -> Result<(), KVStoreError> {
+    /// Not needed/implemented. Maybe this method should go into separate trait?
```
Comment (tizoc) [inline storage/src/backend/kv_store_gced.rs:174]: Can you put ah `TODO: ` prefix here? so that it shows up in searches

Comment (binier) [reply storage/src/backend/kv_store_gced.rs]: Done.

**Comment by tizoc** [comment]
@binier ok, I don't have anything else to comment on this, I don't know about the integrity_test stuff, is there anything that needs to be reviewed by someone there or do you want me to merge this now?

**Comment by binier** [comment]
Yes nothing else to be reviewed.

**Pull request merged**